### ISSUE TITLE
Validation for tf embedding classifier

### ISSUE THREAD

**Issue by jamesmf** [RasaHQ/rasa#1537]
**Feature Request**: The sklearn intent classifier builds `GridSearchCV` into its [Component](https://github.com/RasaHQ/rasa_nlu/blob/master/rasa_nlu/classifiers/sklearn_intent_classifier.py#L160) - thereby giving you an idea of validation accuracy and when you're overfitting.

Is there any equivalent cross validation going on with the embedding classifier? If not, is it possible to refactor the training to have the same `GridSearchCV` wrapper?

**Comment by akelad** [comment]
Thanks for raising this issue, @Ghostvv  will get back to you about it soon.

**Comment by Ghostvv** [comment]
as of now, I think it is not that easy. Also, please take a look at this thread: https://stackoverflow.com/questions/44802939/hyperparameter-tuning-of-tensorflow-model

**Comment by Ghostvv** [comment]
we're working on refactoring tf nlu embedding classifier, and will support calculation of accuracy during training on held out part of provided data
